### ISSUE TITLE
Re-implementation of runBenchmarks

### ISSUE THREAD

**Issue by florianhartig** [florianhartig/DHARMa#239]
runBenchmarks needs an overhaul, as a few problems have accumulated. In particular 

- move to native parallel #52
- fix bug #238 
- improve output #227

Maybe, we could additionally add a few simple output plots, similar to those in the help.

**Comment by florianhartig** [comment]
OK, I think I fixed #227 and #238. @EttnerAndreas, could you test both issues with the code below?

This mainly leaves #52 option, plus the question if we should implement plots. 

```
doCalculations <- function(control = 0){
  
  testData = createData(sampleSize = 200, family = poisson(), overdispersion = control, randomEffectVariance = 0)
  fittedModel <- glm(observedResponse ~ Environment1, data = testData, family = poisson())
  
  out = list()
  
  res <- simulateResiduals(fittedModel = fittedModel, n = 250)
  out$uniformTest = testUniformity(res, plot = F)$p.value  
  out$Dispersion = testDispersion(res, plot = F)$p.value  
  out$DispersionAER = AER::dispersiontest(fittedModel)$p.value
  out$DispPerformance = performance::check_overdispersion(fittedModel)$p_value
  return(unlist(out))
}

# testing a single return
doCalculations(control = 0.3)

dispValues = seq(0,1.2, len = 10)


out = runBenchmarks(doCalculations, nRep = 500,  parallel = F)

# testing calibration

barplot(as.numeric(out$summaries$propSignificant[1,-1]), ylim = c(0,1))
abline(h = 0.05, col = "red")

# testing p-distribution

oPar <- par(mfrow = c(2,2))
for(i in 1:4) hist(out$simulations[,i], main = colnames(out$simulations)[i])
par(oPar)

out = runBenchmarks(doCalculations, controlValues = dispValues , nRep = 100,  parallel = F)

# proportion significant against control 
res = out$summaries$propSignificant
matplot(res$controlValues, res[,-1], type = "l")
legend("bottomright", colnames(res[,-1]), col = 1:4, lty = 1:4)
```

**Comment by EttnerAndreas** [comment]
Code nor working

**Comment by florianhartig** [comment]
OK, but this is your problem, right?

**Comment by florianhartig** [comment]
I mean, it's not a DHARMa problem if the performance package doesn't install on your computer ... maybe an old R version?

**Comment by EttnerAndreas** [comment]
currently running  "R version 4.0.3 (2020-10-10)" 
True, I thought if DHARMa uses 'performance' and I use DHARMa, I get the 'performance' package as well

**Comment by florianhartig** [comment]
Solved up to #52
